src/Microsoft.Identity.Web.OWIN/InternalClaimDetectedException.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Claims;
+
+namespace Microsoft.Identity.Web.OWIN
+{
+    /// <summary>
+    /// The exception that is thrown when an internal ID Token claim used by Microsoft.Identity.Web internally is detected in the user's ID Token.
+    /// </summary>
+    public class InternalClaimDetectedException : Exception
+    {
+        /// <summary>
+        /// Gets or sets the invalid claim.
+        /// </summary>
+        public Claim Claim { get; set; }
```

Comment:
What is the intended usage of this? Are there any privacy/sensitivity concerns?

Reply:
Idea was to add additional information that could be useful for the developer in debugging but can be removed